src/main/java/net/mcreator/blockly/data/BlocklyLoader.java

```diff
@@ -26,41 +31,53 @@ public static void init() {
 		INSTANCE = new BlocklyLoader();
 	}
 
-	private final ExternalBlockLoader procedureBlockLoader;
-	private final ExternalBlockLoader jsonTriggerLoader;
-	private final ExternalBlockLoader aitaskBlockLoader;
-	private final ExternalBlockLoader cmdargsBlockLoader;
-	private final ExternalBlockLoader featureBlockLoader;
+	private static final List<String> builtinCategories = Arrays.asList("other", "apis", "mcelements", "mcvariables",
+			"customvariables", "logicloops", "logicoperations", "math", "text", "time", "advanced", "action", "aiadvanced");
+	private final Map<String, ExternalBlockLoader> blockLoaders;
 	private final ExternalTriggerLoader externalTriggerLoader;
 
 	private BlocklyLoader() {
-		procedureBlockLoader = new ExternalBlockLoader("procedures");
-		aitaskBlockLoader = new ExternalBlockLoader("aitasks");
-		cmdargsBlockLoader = new ExternalBlockLoader("cmdargs");
+		blockLoaders = new LinkedHashMap<>();
+		addBlockLoader("procedures");
+		addBlockLoader("aitasks");
+		addBlockLoader("cmdargs");
+		addBlockLoader("jsontriggers");
+		addBlockLoader("features");
 		externalTriggerLoader = new ExternalTriggerLoader("triggers");
-		jsonTriggerLoader = new ExternalBlockLoader("jsontriggers");
-		featureBlockLoader = new ExternalBlockLoader("features");
 	}
 
-	public ExternalBlockLoader getProcedureBlockLoader() {
-		return procedureBlockLoader;
+	/**
+	 * Create a new {@link ExternalBlockLoader} to load blocks for a specific Blockly panel.
+	 *
+	 * @param name The folder's name where to load files
+	 */
+	public void addBlockLoader(String name) {
+		blockLoaders.put(name, new ExternalBlockLoader(name));
 	}
 
-	public ExternalBlockLoader getAITaskBlockLoader() {
-		return aitaskBlockLoader;
+	/**
+	 * Add a usable category for JSON file blocks that has been created inside the custom XML {@link ToolboxType} file.
+	 * All custom categories used by a {@link ToolboxType} have to be added before blocks are loaded.
+	 *
+	 * @param name The category's name as written inside the XML file (e.g. <i>&lt;custom-thisName/&gt;</i>).
+	 */
+	public static void addBuiltinCategory(String name) {
+		builtinCategories.add(name);

```

Comment:
If you want this to work, you need to wrap current variable initializer with `new ArrayList<>()` (`Arrays.asList()` and `List.of()` alone give you a list of fixed size so this code would throw an UOE).